### PR TITLE
Upgrade puppeteer

### PR DESCRIPTION
**Pull request by cocreature**
We’ve seen a couple of issues in the compatibility tests of the form

```
Error: Protocol error (Runtime.callFunctionOn): Target closed.
```

Looking at the issue tracker in puppeteer this might be fixed in newer
versions and I don’t see why we should stick to a fairly old version
anyway.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: navigator/frontend/package.json [hunk 1]
```diff
@@ -73,14 +73,18 @@
...
-    "redux-thunk": "^2.2.0",
+    "redux-thunk": "~2.2.0",
     "route-parser": "0.0.5",
     "styled-components": "2.1.0",
     "typescript-collections": "1.2.5",
     "uuidjs": "3.5.3",
     "whatwg-fetch": "^2.0.3"
+  },
+  "resolutions": {
+      "**/@types/react": "~16.0.0",
```
Comment (cocreature) [inline navigator/frontend/package.json:87]: The fact that we use `@types/react` 16.x but react 15.x seems super sketchy but that’s what we’ve been doing before as well. This doesn’t change anything in that regard.

File: navigator/frontend/package.json [hunk 2]
```diff
@@ -27,11 +27,11 @@
     "@types/color": "1.0.3",
     "@types/deep-equal": "1.0.0",
     "@types/lodash": "4.14.76",
-    "@types/react": "^16.0.5",
+    "@types/react": "~16.0.5",
     "@types/react-autosuggest": "8.0.1",
     "@types/react-dom": "^15.5.0",
     "@types/react-overlays": "0.8.4",
-    "@types/react-redux": "^5.0.8",
+    "@types/react-redux": "^5.0.8 <5.0.13",
```
Comment (cocreature) [inline navigator/frontend/package.json:34]: I love breaking changes in minor releases.

**Comment by rohanjr** [comment]
Failure on mac:
```
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/nix/store/pyi0rf2wm668yndfxghiph8wfbbavxc7-nodejs-12.16.2/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
```
And on Windows compatibility after a `yarn install`:
```
error puppeteer@3.1.0: The engine "node" is incompatible with this module. Expected version ">=10.18.1". Got "10.16.3"
error Found incompatible module.
```

**Comment by cocreature** [comment]
The mac failure should be fixed by removing `node-live-server`. That’s why I was asking on Slack whether anyone uses it. Still looking into the Windows failure.

**Pull request merged**